### PR TITLE
[IOPAY-485] Delete rate limit for api for io-backend

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- Delete rate limit for api for io-backend
- Ad ip-filter for checkout auth
- add `detail_v2` in checkout auth policy

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

**Review by pasqualedevita — COMMENTED** [review]
I suggest a different approach:
1. create rate limit groups in apim (for example checkout-rate-limit-300, checkout-rate-limit-600, checkout-rate-limit-1000, checkout-rate-no-limit)
2. add io-backend to group
3. update policy like https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/apim/api_management_product_io-services-api/policy.xml

**Comment by infantesimone** [comment]
> I suggest a different approach:
> 
> 1. create rate limit groups in apim (for example checkout-rate-limit-300, checkout-rate-limit-600, checkout-rate-limit-1000, checkout-rate-no-limit)
> 2. add io-backend to group
> 3. update policy like https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/apim/api_management_product_io-services-api/policy.xml

How can i link the `io-backend` subscription key with the group?

**Comment by pasqualedevita** [comment]
> > I suggest a different approach:
> > 
> > 1. create rate limit groups in apim (for example checkout-rate-limit-300, checkout-rate-limit-600, checkout-rate-limit-1000, checkout-rate-no-limit)
> > 2. add io-backend to group
> > 3. update policy like https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/apim/api_management_product_io-services-api/policy.xml
> 
> How can i link the `io-backend` subscription key with the group?

we need to add also a user

**Comment by infantesimone** [comment]
> > > I suggest a different approach:
> > > 
> > > 1. create rate limit groups in apim (for example checkout-rate-limit-300, checkout-rate-limit-600, checkout-rate-limit-1000, checkout-rate-no-limit)
> > > 2. add io-backend to group
> > > 3. update policy like https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/apim/api_management_product_io-services-api/policy.xml
> > 
> > 
> > How can i link the `io-backend` subscription key with the group?
> 
> we need to add also a user

So:
-  new io-backend user (from portal) with new subscription key linked to this user. 
-  Then I bind the group to the user.

**Comment by pasqualedevita** [comment]
> > > > I suggest a different approach:
> > > > 
> > > > 1. create rate limit groups in apim (for example checkout-rate-limit-300, checkout-rate-limit-600, checkout-rate-limit-1000, checkout-rate-no-limit)
> > > > 2. add io-backend to group
> > > > 3. update policy like https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/apim/api_management_product_io-services-api/policy.xml
> > > 
> > > 
> > > How can i link the `io-backend` subscription key with the group?
> > 
> > 
> > we need to add also a user
> 
> So:
> 
> * new io-backend user (from portal) with new subscription key linked to this user.
> * Then I bind the group to the user.

yes perfect. U can link subscription and user without delete the subscription

File: src/api/checkout/checkout_payment_activations_auth/v1/_base_policy.xml.tpl [hunk 1]
```diff
@@ -15,7 +15,12 @@
       </cors>
       <base />
       <set-backend-service base-url="{{pagopa-appservice-proxy-url}}" />
-      <rate-limit-by-key calls="150" renewal-period="10" counter-key="@(context.Request.Headers.GetValueOrDefault("X-Forwarded-For"))" />
+      <ip-filter action="allow">
+        <address>io-backend</address> <!-- TODO-->
```
Comment (infantesimone) [inline src/api/checkout/checkout_payment_activations_auth/v1/_base_policy.xml.tpl:19]: @pasqualedevita could you tell me the IP value of io-backend?

**Review by pasqualedevita — APPROVED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**